ds/ds.go

```diff
@@ -275,6 +276,56 @@ func Lookup(query dsQuery) (string, string, error) {
 	return e.IPs[0].String(), strconv.Itoa(e.Port), nil
 }
 
+// LookupResult is returned over a channel, and has either a result or an error.
+type LookupResult struct {
+	Entry *dnssd.BrowseEntry
+	Error error
+}
+
+// LookupChan returns a chan of *LookupResult and a cancel function.
+// The cancel function can be called when the desired number of results has
+// been received; or for any other reason in fact.
+func LookupChan(query dsQuery) (chan *LookupResult, context.CancelFunc) {
```

Comment:
Just pointing out "Rethinking Classical Concurrency Patterns" by bcmills - (video: https://www.youtube.com/watch?v=5zXAHh5tJqQ slides:   https://drive.google.com/file/d/1nPdvhB0PutEJzdCq5ms6UI58dp50fcAN/view).    Exposing channels as part of the public API of this package is risky and ambiguous.  Whose job is it to close the channel - the caller or the callee **(explicitely: can you please document the expected behavior (who will close the channel) in the docstring? thanks!)?**

Reply:
done. Thanks.

The reason for the chan is that one might overlap getting lookup results and initiating the cpu connections. This is a common pattern in HPC. 

In this case, I doubt the lookup will take that long, so I'm okay with the changes.